plugins/lookup/random_string.py

```diff
@@ -0,0 +1,217 @@
+      - Generates random string based upon the given constraints.
+    options:
+      length:
+        description: The length of the string.
+        default: 8
+        type: int
+      upper:
+        description:
+        - Include uppercase alphabets in the string.
+        default: true
+        type: bool
+      lower:
+        description:
+        - Include lowercase alphabets in the string.
+        default: true
+        type: bool
+      numbers:
+        description:
+        - Include numbers in the string.
+        default: true
+        type: bool
+      special:
+        description:
+        - Include special characters in the string.
+        - Special characters are taken from Python standard library C(string).
+        default: true
+        type: bool
+      min_numeric:
+        description:
+        - Minimum number of numeric characters in the string.
+        - If set, overrides I(numbers=false).
+        default: 0
+        type: int
+      min_upper:
+        description:
+        - Minimum number of uppercase alphabets in the string.
+        - If set, overrides I(upper=false).
+        default: 0
+        type: int
+      min_lower:
+        description:
+        - Minimum number of lowercase alphabets in the string.
+        - If set, overrides I(lower=false).
+        default: 0
+        type: int
+      min_special:
+        description:
+        - Minimum number of special character in the string.
+        default: 0
+        type: int
+      override_special:
+        description:
+        - Overide a list of special characters to use in the string.
+        - If set I(min_special) should be set to a non-default value.
+        type: str
+      override_all:
+        description:
+        - Override all values of I(numbers), I(upper), I(lower), and I(special) with
+          the given list of characters.
+        type: str
+      base64:
+        description:
+        - Returns base64 encoded string.
+        type: bool
+        default: false
+"""
+
+EXAMPLES = r"""
+- name: Generate random string
+  ansible.builtin.debug:
+    var: lookup('community.general.random_string')
+  # Example result: 'DeadBeeF'
+
+- name: Generate random string with length 12
+  ansible.builtin.debug:
+    var: lookup('community.general.random_string', length=12)
+  # Example result: 'Uan0hUiX5kVG'
+
+- name: Generate base64 encoded random string
+  ansible.builtin.debug:
+    var: lookup('community.general.random_string', base64=True)
+  # Example result: 'NHZ6eWN5Qk0='
+
+- name: Generate a random string with 1 lower, 1 upper, 1 number and 1 special char (atleast)
+  ansible.builtin.debug:
+    var: lookup('community.general.random_string', min_lower=1, min_upper=1, min_special=1, min_numeric=1)
+  # Example result: '&Qw2|E[-'
+
+- name: Generate a random string with all lower case characters
+  debug:
+    var: query('community.general.random_string', upper=false, numbers=false, special=false)
+  # Example result: 'exolxzyz'
```

Comment:
```suggestion
  # Example result: ['exolxzyz']
```
`query` returns a list with one string in it. Either change the example to use `lookup`, or add `[0]` at the end, or adjust this comment. (Same below.)